typescript/api-client/projects/vcd/plugin-builders/src/lib/base/index.ts

```diff
@@ -1,25 +1,65 @@
+import * as fs from 'fs';
+import * as path from "path";
+import * as ZipPlugin from 'zip-webpack-plugin';
 import {
   BrowserBuilder,
   NormalizedBrowserBuilderSchema
 } from '@angular-devkit/build-angular';
 import { Path, virtualFs } from '@angular-devkit/core';
-import * as fs from 'fs';
 import { Observable } from 'rxjs';
-
 import { BuilderConfiguration, BuildEvent } from '@angular-devkit/architect';
 import { tap } from 'rxjs/operators';
-import * as ZipPlugin from 'zip-webpack-plugin';
+import { extractExternalRegExps, splitVendorsIntoChunks, processManifestJsonFile, VCD_CUSTOM_LIB_SEPARATOR, takeChunkPackageJson } from "./utilites";
+import { LibrariesConfig, ExtensionManifest } from "./interfaces";
+import { ConcatWebpackPlugin } from "./concat";
 
-interface PluginBuilderSchema extends NormalizedBrowserBuilderSchema {
+export interface PluginBuilderSchema extends NormalizedBrowserBuilderSchema {
+	vcdVersion: string;
   /**
-   * A string of the form `path/to/file#exportName` that acts as a path to include to bundle
+   * A string of the form `path/to/file#exportName`
+	 * that acts as a path to include to bundle.
    */
-  modulePath: string;
+	modulePath: string;
+	/**
+	 * List of external libraries defined by the user.
+	 */
+	externalLibs: string[];
+	/**
+	 * Will disable the default external libraries,
+	 * allowing the user to define his own thanks to externalLibs property.
+	 */
+	ignoreDefaultExternals: boolean;
+	/**
+	 * List of libraries determining thier version, scope and file name (location). 
+	 */
+	librariesConfig: LibrariesConfig;
+}
+
+export const defaultExternals = {
+	common: [
+		/^@angular\/.+$/,
+		/^@ngrx\/.+$/,
+		/^@vcd\/common$/,
+		/^@vcd-ui\/common$/,
+		{
+			reselect: 'reselect'
+		}
+	],
+	["9.7-10.0"]: [
+		/^rxjs(\/.+)?$/,
+		/^@clr\/.+$/,
+		{
+			'clarity-angular': 'clarity-angular',
+		}
+	]
 }
+
 export default class PluginBuilder extends BrowserBuilder {
   private options: PluginBuilderSchema;
 
   private entryPointPath: string;
+	private entryPointOriginalContent: string;
```

Comment:
It seems there are formatting issues. Seems you are using tabs instead of spaces.